Modules/arraymodule.c

```diff
@@ -46,18 +48,28 @@ typedef struct arrayobject {
     Py_ssize_t ob_exports;  /* Number of exported buffers */
 } arrayobject;
 
-static PyTypeObject Arraytype;
-
 typedef struct {
     PyObject_HEAD
     Py_ssize_t index;
     arrayobject *ao;
     PyObject* (*getitem)(struct arrayobject *, Py_ssize_t);
 } arrayiterobject;
 
-static PyTypeObject PyArrayIter_Type;
+typedef struct {
+    PyTypeObject *ArrayType;
+    PyTypeObject *ArrayIterType;
+} array_state;
 
-#define PyArrayIter_Check(op) PyObject_TypeCheck(op, &PyArrayIter_Type)
+static array_state *
+get_array_state(PyObject *module)
+{
+    return (array_state *)PyModule_GetState(module);
+}
+
+#define get_array_state_by_type(tp) \
+    (get_array_state(_PyType_GetModuleByDef(tp, &arraymodule)))
```

Comment:
"find_array_state_by_type" may be a better name, see discussion at:
https://github.com/python/cpython/pull/22835#issuecomment-721879674

Reply:
I agree; thanks!

Reply:
Adjusted the naming in 72434ad